cli/core/pkg/command/discovery_source.go

```diff
@@ -193,25 +193,7 @@ var deleteDiscoverySourceCmd = &cobra.Command{
 	RunE: func(cmd *cobra.Command, args []string) (err error) {
 		discoveryName := args[0]
 
-		// Acquire tanzu config lock
-		configlib.AcquireTanzuConfigLock()
-		defer configlib.ReleaseTanzuConfigLock()
-
-		cfg, err := configlib.GetClientConfigNoLock()
-		if err != nil {
-			return err
-		}
-		if cfg.ClientOptions == nil || cfg.ClientOptions.CLI == nil {
-			return fmt.Errorf("discovery %q unknown", discoveryName)
-		}
-
-		newDiscoverySources, err := deleteDiscoverySource(cfg.ClientOptions.CLI.DiscoverySources, discoveryName)
-		if err != nil {
-			return err
-		}
-
-		cfg.ClientOptions.CLI.DiscoverySources = newDiscoverySources
-		err = configlib.StoreClientConfig(cfg)
+		err = configlib.DeleteCLIDiscoverySource(discoveryName)
```

Comment:
The intention is to support deleting the first matching source by name? If so it may be a good idea to add a comment to deleteCLIDiscoverySource near where getCLIDiscoverySource is called.

Reply:
since discovery source are unique by name itself. There should only be one entry matching with the name.

Reply:
So it is not intentional, but the code does not error out on config with multiple sources of the same name. Which is probably a not a bad thing, since it means the implementation can still be used to correct the discovery sources should the file somehow reach that state. 
I think it warrants at least a comment say at getCLIDiscoverySource that we are making an assumption that there should only be one source with the name, hence we are returning the first match.

Reply:
added the description on GetCLIDiscoverySource()